packages/server/src/workflows/triggers.js

```diff
@@ -4,7 +4,79 @@ const InMemoryQueue = require("./queue/inMemoryQueue")
 
 let workflowQueue = new InMemoryQueue()
 
-async function queueRelevantWorkflows(event, eventType) {
+const FAKE_STRING = "TEST"
+const FAKE_BOOL = false
```

Comment:
What are these fake values for?

Reply:
Basically, something that will need to be solved in the future, but right now we are using the trigger endpoint simply for the builder to be able to test their workflows.

As we are using the trigger endpoint right now to test there is a problem that the `RECORD_SAVED` and `RECORD_DELETED` triggers generate a record as an output when they complete, this can be used as inputs for future stages - when the workflow is triggered from the endpoint the builder does not supply any parameters - it just triggers it, therefore any trigger outputs used in the workflow will be null and the workflow could end up performing differently to how it will perform when it is live.

We thought about whether or not to handle this edge case, I thought with time constraints it would be better to take the work out of the builder and simplify the test experience for the user and simply have the server populate a fake record for the model that was being triggered so that the workflow would perform similarly to what it would in production (just with test data).

Open to input on this, its not the best manner of handling it but I feel something needs to be done for it, otherwise a test workflow button isn't the biggest amount of use, you're better actually going and triggering the workflow (which is also a totally fine way of doing it but I think for some users a simple test workflow button is good).

As a sidenote: In the near future (possibly pre-launch) we may want to allow the user to actually configure a trigger which is the endpoint itself, using the POST body as the output parameters of the trigger which can be used as inputs for future workflow steps. Its a relatively small piece of work to do this, simply creating a new trigger definition for an API endpoint - just need to think about how we would validate this for external apps and how we validate its inputs/outputs.

Reply:
Ah, see what you mean. That's perfect - I'd imagine we will complete most of that work you mentioned as part of https://github.com/Budibase/budibase/issues/495. Anything that pertains to the external execution of workflows via HTTP would make sense to be solved there.

This works great for testing workflows - I actually think this is a good way to identify the test record when it's created so you can just delete it again.